user/dao/user-dao.go

```diff
@@ -2,31 +2,52 @@ package dao
 
 import (
 	"database/sql"
+	"errors"
 
 	// pq acts as the driver for SQL requests
 	_ "github.com/lib/pq"
 )
 
+// TODO: This should come from a configuration file
+// Host matches the container name in docker-compose.yml
+// https://docs.docker.com/compose/networking/
+const connStr = "user=postgres dbname=postgres host=user-db sslmode=disable"
+
 // UserGetResponse returns all the information stored about a user
 type UserGetResponse struct {
 	ID   int
 	Name string
 }
 
-// GetUser returns the information about a user stored for a given ID
-func GetUser(id int64) (*UserGetResponse, error) {
-	// Host matches the container name in docker-compose.yml
-	// https://docs.docker.com/compose/networking/
-	connStr := "user=postgres dbname=postgres host=user-db sslmode=disable"
+// UserCreateRequest contains the information required to create a new user
+type UserCreateRequest struct {
+	Name string `valid:"type(string),required,stringlength(2|255)"`
```

Comment:
appreciate this isn't great for generation, but is very concise. We could break it into method calls if we thing that is easier to generate?